### PR TITLE
Add the ability to query ALB logs with Athena

### PR DESCRIPTION
**Pull request by rbreslow**
## Overview

This PR adds the ability to query ALB logs with Athena by codifying the following AWS guides:

- [Access logs for your Application Load Balancer → Enable access logging](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html#enable-access-logging)
- [Querying Application Load Balancer Logs → Create ALB Table](https://docs.aws.amazon.com/athena/latest/ug/application-load-balancer-logs.html#create-alb-table)

Connects #428 

### Checklist

- [x] Description of PR is in an appropriate section of `CHANGELOG.md` and grouped with similar changes, if possible

## Testing Instructions

See that the results of this basic query look right (`client_ip` has been redacted).

```sql
SELECT *
FROM stg_alb_logs LIMIT 5;
```

[d2063156-1045-4c1a-86a8-238bef2c1db2.csv.zip](https://github.com/PublicMapping/districtbuilder/files/5270179/d2063156-1045-4c1a-86a8-238bef2c1db2.csv.zip)

### CODE REVIEW

File: deployment/terraform/iam.tf [hunk 1]
```diff
@@ -53,3 +53,27 @@ resource "aws_iam_role_policy" "scoped_email_sending" {
   role   = aws_iam_role.ecs_task_role.name
   policy = data.aws_iam_policy_document.scoped_email_sending.json
 }
+
+data "aws_elb_service_account" "main" {}
```
Comment (rbreslow) [inline deployment/terraform/iam.tf:57]: This is the same [approach](https://github.com/azavea/raster-foundry-deployment/blob/4ca5f0d0132696b410d5f422e9e98e2e41e06c6f/core/terraform/iam.tf#L270-L279) we take in Raster Foundry. According to the AWS docs, the bucket policy must still reference the `arn:aws:iam::elb-account-id:root` principal.

File: deployment/terraform/athena.tf [hunk 2]
```diff
@@ -0,0 +1,196 @@
+locals {
+  short_environment = var.environment == "Staging" ? "stg" : var.environment == "Production" ? "prd" : lower(var.environment)
+}
```
Comment (rbreslow) [inline deployment/terraform/athena.tf:3]: Translates to the following pseudocode:

```lua
if environment == "Staging" then
    return "stg"
elseif environment == "Production" then
   return "prd"
else
   return string.lower(environment)
end

File: deployment/terraform/app.tf [hunk 3]
```diff
@@ -30,11 +30,24 @@ resource "aws_lb" "app" {
...
+  }
+
   tags = {
     Name        = "alb${var.environment}App"
     Project     = var.project
     Environment = var.environment
   }
+
+  # In order to enable access logging, the ELB service account needs S3 access.
+  # This is a "hidden" dependedncy that Terraform cannot automatically infer, so
```
Comment (hectcastro) [inline deployment/terraform/app.tf:46]:
```suggestion
  # This is a "hidden" dependency that Terraform cannot automatically infer, so
```

File: deployment/terraform/athena.tf [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+# https://docs.aws.amazon.com/athena/latest/ug/application-load-balancer-logs.html
+resource "aws_glue_catalog_table" "alb_logs" {
```
Comment (hectcastro) [inline deployment/terraform/athena.tf:6]: Nice to see an example of this.

**Review by hectcastro — APPROVED** [review]
Table looks good. Nice to have an example of an Athena table defined in Terraform. 👍

**Pull request merged**